tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -434,6 +434,150 @@ object SQLPlanParser extends Logging {
     parsedExpressions.distinct.toArray
   }
 
+   // This parser is used for BroadcastHashJoin, ShuffledHashJoin and SortMergeJoin
+   def parseEquijoinsExpressions(exprStr: String): (Array[String], Boolean) = {
+     // ShuffledHashJoin [name#11, CEIL(DEPT#12)], [name#28, CEIL(DEPT_ID#27)], Inner, BuildLeft
+     // SortMergeJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner
+     // BroadcastHashJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner,
+     // BuildRight, false
+     val parsedExpressions = ArrayBuffer[String]()
+     val pattern = """\[([\w#, +*\\\-\.<>=\`\(\)]+\])""".r
```

Comment:
regex pattern creation typically should be outside method calls to amortize the creation cost.

Reply:
Thanks for the suggestion. I tend to declare them where it is used but declaring outside the method calls in this case makes sense. Updated in [670c8b1](https://github.com/NVIDIA/spark-rapids/pull/6796/commits/670c8b148bca5ad33e8b230127e787f535ece105)